File: Pasha and Stick.java
//https://codeforces.com/contest/610/problem/A
import java.util.*;
public class Main {
	public static void main(String[] args) {
		Scanner sc=new Scanner(System.in);
		int n=sc.nextInt(),r;
		if(n%2==1||n<6) r=0;
		else r=(n-2)/4;
		System.out.println(r);
	}
}

File: Pasha and Stick.cpp
//https://codeforces.com/contest/610/problem/A
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(NULL);cout.tie(NULL);
    int n;
    cin>>n;
    if(n%2||n<6) cout<<0<<endl;
    else cout<<((n-2)/4)<<endl;
}